File: app/src/main/java/me/tysheng/xishi/inter/DataCallback.java
package me.tysheng.xishi.inter;

import android.support.v7.util.DiffUtil;
import android.text.TextUtils;

import java.util.ArrayList;

import me.tysheng.xishi.bean.Album;

/**
 * Created by Sty
 * Date: 16/9/21 13:18.
 */

public class DataCallback extends DiffUtil.Callback {
    private ArrayList<Album> oldList, newList;

    public DataCallback(ArrayList<Album> oldList, ArrayList<Album> newList) {
        this.oldList = oldList;
        this.newList = newList;
    }

    @Override
    public int getOldListSize() {
        return oldList.size();
    }

    @Override
    public int getNewListSize() {
        return newList.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return TextUtils.equals(oldList.get(oldItemPosition).id, newList.get(newItemPosition).id);
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        return TextUtils.equals(oldList.get(oldItemPosition).url, newList.get(newItemPosition).url);
    }
}

File: app/src/main/java/me/tysheng/xishi/inter/DataCallback.kt
package me.tysheng.xishi.inter

import android.support.v7.util.DiffUtil
import android.text.TextUtils

import java.util.ArrayList

import me.tysheng.xishi.bean.Album

/**
 * Created by Sty
 * Date: 16/9/21 13:18.
 */

class DataCallback(private val oldList: ArrayList<Album>, private val newList: ArrayList<Album>) : DiffUtil.Callback() {

    override fun getOldListSize(): Int {
        return oldList.size
    }

    override fun getNewListSize(): Int {
        return newList.size
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return TextUtils.equals(oldList[oldItemPosition].id, newList[newItemPosition].id)
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return TextUtils.equals(oldList[oldItemPosition].url, newList[newItemPosition].url)
    }
}
